src/devices/vdml_usd.cpp

```diff
@@ -71,29 +71,30 @@ std::vector<std::string> list_files(const char* path, char* buffer, int32_t len)
 		}
 		return files;
 	}
+
 	// Parse buffer given call successful, split by '/n'
-	// Store char * buffer as std::string in str
-	// Store delimiter in std::string, '\n'
-	std::string str(buffer);
-	std::string delimiter = "\n";
+	std::string_view str(buffer);
 
-	// position to keep track of position in str
+	// delimter_pos is the position of the delimiter '\n'
+	// index of which character to start substr from
 	// file_name used to store each file name
-	size_t position = 0;
-	std::string file_name;
+	size_t delimiter_pos, index = 0;
+	std::string_view file_name;
 
 	// Loop until delimiter '\n' can not be found anymore
-	while ((position = str.find(delimiter)) != std::string::npos) {
+	while ((delimiter_pos = str.find('\n', index)) != std::string::npos) {
 		// file_name is the string from the beginning of str to the first '\n', excluding '\n'
-		file_name = str.substr(0, position);
+		file_name = std::string_view(str.data() + index, delimiter_pos - index);
 		// Add token to files vector
-		files.push_back(file_name);
-		// Erase file_name from str, would be pos + 1 if we wanted to include '\n'
-		str.erase(0, position + 1);
-	}
+		files.emplace_back(file_name);
```

Comment:
Since it happens via an implicit conversion, it would be nice to have a comment here noting that this is the point where the string containing the file name is copied into its own `std::string`, since everything before this was just `std::string_view`s over the pre-existing `buffer`.